File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/presentation/settings/SetTimerPageAdapter.java
package com.igorganapolsky.vibratingwatchapp.presentation.settings;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import com.igorganapolsky.vibratingwatchapp.presentation.settings.step.SetTimerRepeatFragment;
import com.igorganapolsky.vibratingwatchapp.presentation.settings.step.SetTimerTimeFragment;
import com.igorganapolsky.vibratingwatchapp.presentation.settings.step.SetTimerVibrationFragment;

public class SetTimerPageAdapter extends FragmentStatePagerAdapter {

    private final Fragment[] fragments = {
        new SetTimerTimeFragment(),
        new SetTimerVibrationFragment(),
        new SetTimerRepeatFragment()
    };

    SetTimerPageAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        return fragments[i];
    }

    @Override
    public int getCount() {
        return fragments.length;
    }
}

File: app/src/main/kotlin/com/igorganapolsky/vibratingwatchapp/presentation/timer_creation_screen/SetTimerPageAdapter.kt
package com.igorganapolsky.vibratingwatchapp.presentation.timer_creation_screen

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.igorganapolsky.vibratingwatchapp.presentation.timer_creation_screen.step.SetTimerRepeatFragment
import com.igorganapolsky.vibratingwatchapp.presentation.timer_creation_screen.step.SetTimerTimeFragment
import com.igorganapolsky.vibratingwatchapp.presentation.timer_creation_screen.step.SetTimerVibrationFragment

internal class SetTimerPageAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    private val fragments =
        arrayOf(SetTimerTimeFragment(), SetTimerVibrationFragment(), SetTimerRepeatFragment())

    override fun getItem(i: Int): Fragment {
        return fragments[i]
    }

    override fun getCount(): Int {
        return fragments.size
    }
}
